File: programmers/jihwjeon/138476.py
def solution(k, tangerine):
    answer = 0
    d = {}
    
    for t in tangerine:
        if t not in d:
            d[t] = 1
        else:
            d[t] += 1
            
    d = sorted(d.items(), key=lambda item: item[1], reverse=True)
    
    tmp = 0
    for v, cnt in d:
        tmp += cnt
        answer += 1
        if tmp >= k:
            break
    return answer

File: programmers/yuhwang/138476.swift
import Foundation

func solution(_ k:Int, _ tangerine:[Int]) -> Int {
    var counts: [Int: Int] = [:]
    var K = tangerine.count - k
    for i in tangerine {
        if let c = counts[i] {
            counts[i] = c + 1
        } else {
            counts[i] = 1
        }
    }
    let count = Array(counts).sorted { $0.value < $1.value }
    var idx = 0
    while K > 0 {
        if K >= count[idx].value {
            K -= count[idx].value
            idx += 1
        } else {
            K = 0
            break
        }
    }
    return count.count - idx
}
